src/js/relevant.js

```diff
@@ -65,29 +108,50 @@ export default {
                 return;
             }
 
-            /*
-             * Check if the (calculate without form control) node is part of a repeat that has no instances
-             */
-            const pathParts = p.path.split('/');
-            if (pathParts.length > 3) {
-                const parentPath = pathParts
-                    .splice(0, pathParts.length - 1)
-                    .join('/');
-                const parentGroups = [
-                    ...this.form.view.html.querySelectorAll(
-                        `.or-group[name="${parentPath}"],.or-group-data[name="${parentPath}"]`
-                    ),
-                ]
-                    // now remove the groups that have a repeat-info child without repeat instance siblings
-                    .filter(
-                        (group) =>
-                            getChild(group, '.or-repeat') ||
-                            !getChild(group, '.or-repeat-info')
-                    );
-                // If the parent doesn't exist in the DOM it means there is a repeat ancestor and there are no instances of that repeat.
-                // Hence that relevant does not need to be evaluated (and would fail otherwise because the context doesn't exist).
-                if (parentGroups.length === 0) {
-                    return;
+            const { repeatIndex } = options;
```

Comment:
Performance changes:
- Don't do repeat logic if there are no repeats
- Look in the caches before doing any logic

Discovered as part of making things correct.